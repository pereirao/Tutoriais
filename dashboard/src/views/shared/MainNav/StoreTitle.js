// @flow

import React from 'react';
import {connect} from 'react-redux';
import {session_helpers, session_selectors} from '../../../business/session';

import style_sheet from './MainNav.module.css';

const {supplierOpenMessage} = session_helpers;
const {supplierName, shippingMethods} = session_selectors;

type StoreTitleProps = { supplier_name?: string, delivery_info?: string };
export const StoreTitle = ({supplier_name, delivery_info}: StoreTitleProps) => (
  <div className={style_sheet.storeTitleWrapper}>
    <h1 className={style_sheet.storeTitle}>{supplier_name}</h1>
    <p className={style_sheet.storeSubTitle}>{delivery_info}</p>
  </div>
);

const StoreTitleSTP = state => ({
  supplier_name: supplierName(state),
  delivery_info: supplierOpenMessage(shippingMethods(state))
});
const StoreTitleContainer = connect(StoreTitleSTP)(StoreTitle);

export default StoreTitleContainer;
