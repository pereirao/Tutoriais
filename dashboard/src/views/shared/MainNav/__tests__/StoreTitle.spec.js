import React from 'react';

import { StoreTitle } from '../StoreTitle';

it('renders StoreTitle correctly', () => {
  expect(shallow(<StoreTitle supplier_name="House of Liquids" delivery_info="open daily from dusk til dawn" />)).toMatchSnapshot();
});
