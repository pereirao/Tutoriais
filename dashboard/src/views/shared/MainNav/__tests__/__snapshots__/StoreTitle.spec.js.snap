// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders StoreTitle correctly 1`] = `
<div
  className="storeTitleWrapper"
>
  <h1
    className="storeTitle"
  >
    House of Liquids
  </h1>
  <p
    className="storeSubTitle"
  >
    open daily from dusk til dawn
  </p>
</div>
`;
